hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/SimpleMockNodeManager.java

```diff
@@ -87,7 +87,8 @@ public void setPipelines(DatanodeDetails dd, int count) {
    *         Inservice and Healthy NodeStatus.
    */
   @Override
-  public NodeStatus getNodeStatus(DatanodeDetails datanodeDetails) {
+  public NodeStatus getNodeStatus(DatanodeDetails datanodeDetails)
+      throws NodeNotFoundException {
```

Comment:
Shouldn't this method really throw `NodeNotFoundException` if `dni` is not found, instead of returning a healthy status, to more closely reflect actual `NodeManager` behavior?

Reply:
Yea it would make more sense for this method do thow NodeNotFoundException, however, I cheated somewhat when using this class in an earlier patch within TestReplicationManager. Replication manager only calls the "getNodeStatus()" method of the NodeManager, and to avoid having to change too many existing tests I defaulted this to return "Healthy + Inservice" for any node not registered so the existing tests passed without modification.

I have an existing Jira HDDS-2673 to refactor and merge MockNodeManager and SimpleMockNodeManager into one, so perhaps we could improve this issue as part of that?

Reply:
> perhaps we could improve this issue as part of that?

Sure, that's fine, too.

Reply:
I added a comment to HDDS-2673 to remind about this when I get to working on it.